src/bin/common/parse_command.ml

```diff
@@ -1216,13 +1256,81 @@ let parse_theory_opt =
     Arg.(value & flag & info ["tighten-vars"] ~docs ~doc) in
 
   let use_fpa =
-    let doc = "Enable support for floating-point arithmetic." in
-    Arg.(value & flag & info ["use-fpa"] ~docs ~doc) in
+    let doc = "Floating-point builtins are always enabled and this option has
+    no effect anymore. It will be removed in a future version." in
+    let deprecated = "this option is always enabled" in
+    Arg.(value & flag & info ["use-fpa"] ~docs ~doc ~deprecated) in
+
+  let theories =
+    let theory_enum =
+      Preludes.all
+      |> List.map (fun t -> Format.asprintf "%a" Preludes.pp t, t)
+    in
+    let theory = Arg.enum theory_enum in
+    let enable_preludes =
+      let doc =
+        Format.asprintf "Enable theory preludes, multiple comma-separated values
+        are supported. $(docv) must be %s."
+          (Arg.doc_alts_enum theory_enum)
+      in
+      let docv = "THEORY" in
+      Arg.(
+        value & opt (list theory) [] & info ["enable-preludes"] ~docs ~doc ~docv
+      )
+    and disable_preludes =
```

Comment:
In my opinion, it would nicer to adda common option to disable theory. We could type --disable-theory nameofprelude, adt, ...

Reply:
Agreed but I think it is out of scope for this PR (and probably for 2.5.0 ? I opened #657 ; but not sure we should add it to 2.5.0 at that point)

What I could do is rename this option `--disable-theories` and `--enable-theories` etc. but "lie" and only include the prelude theories in there. Then we can move the other options to the list later. Does that sound like an acceptable compromise?

Reply:
I agree. Let's merge your solution for the 2.5.0 :)